monai/data/dataset.py

```diff
@@ -246,6 +247,28 @@ def __init__(
                 self.cache_dir.mkdir(parents=True, exist_ok=True)
             if not self.cache_dir.is_dir():
                 raise ValueError("cache_dir must be a directory.")
+        self.set_transform_hash()
```

Comment:
Thanks for the good feature.
I am afraid it may affect the performance of PerisistentDataset, especially when not JSON serializable.
Could you please help make it optional and default to True?
CC @wyli @ericspod 

Thanks.

Reply:
agreed, @rijobro could you please make it optional? then I'll merge this one soon. Thanks!